frontend/tests/controllers/CourseCreateTest.php

```diff
@@ -205,6 +210,115 @@ public function testCreateSchoolAssignment() {
         $this->assertEquals($points, $problems[0]->points);
     }
 
+    /**
+     * A PHPUnit data provider for the test with problems in a course.
+     *
+     * @return list<array{0: list<int|null>, 1: int, 2: bool}>
+     */
+    public function assignmentWithProblemsAndPointsValueProvider(): array {
+        return [
+            [[null, null, null], 300, false],
+            [[100, 80, 20], 200, false],
+            [[50, null, 20], 170, false],
+            [[50, null, '80'], 230, false],
+            [['wrong_value', null, '80'], 230, true],
+        ];
+    }
+
+    /**
+     * @param list<int|null>
+     *
+     * @dataProvider assignmentWithProblemsAndPointsValueProvider
+     */
+    public function testCreateSchoolAssignmentWithProblems(
+        array $problemPoints,
+        int $expectedTotalPoints,
+        bool $expectedError
+    ) {
+        // Create a test course
+        [
+            'user' => $user,
+            'identity' => $identity,
+        ] = \OmegaUp\Test\Factories\User::createUser();
+        $courseAlias = \OmegaUp\Test\Utils::createRandomString();
+        $login = self::login($identity);
+
+        // Call api
+        $response = \OmegaUp\Controllers\Course::apiCreate(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'name' => \OmegaUp\Test\Utils::createRandomString(),
+                'alias' => $courseAlias,
+                'description' => \OmegaUp\Test\Utils::createRandomString(),
+                'start_time' => (\OmegaUp\Time::get() + 60),
+                'finish_time' => (\OmegaUp\Time::get() + 120)
+            ])
+        );
+        $this->assertEquals('ok', $response['status']);
+
+        // Create 3 problems
+        $numberOfProblems = count($problemPoints);
+        $pointsTotal = 0;
+        $problemsData = [];
+        foreach ($problemPoints as $i => $points) {
```

Comment:
nit:

```suggestion
        foreach ($problemPoints as $points) {
```

el índice ya no se va a necesitar.

Reply:
Se quita